src/app/api/comments/route.ts

```diff
@@ -0,0 +1,146 @@
+import { NextResponse, type NextRequest } from "next/server";
+import { createCommentSchema } from "./schema";
+
+const prisma = new PrismaClient();
+
+export async function POST(req: NextRequest) {
+  try {
+    console.log("Received request:", req);
+    const requestBody = await req.json();
+    console.log("Request body:", requestBody);
+
+    const parsed = createCommentSchema.safeParse(requestBody);
+    console.log("Validation result:", parsed);
+
+    if (!parsed.success) {
+      console.error("Validation error:", parsed.error);
+      return new NextResponse(JSON.stringify({ error: parsed.error }), {
+        status: 400,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+    }
+
+    const commentData = parsed.data;
+
+    const createdComment = await prisma.comment.create({
+      data: {
+        content: commentData.content,
+        quote: commentData.quote,
+        // parentCommentId: commentData.parentCommentId,
+        userId: commentData.userId,
+        storyId: commentData.storyId,
+      },
+    });
+
+    return new NextResponse(
+      JSON.stringify({
+        message: "Comment created successfully",
+        commentData: createdComment,
+      }),
+      {
+        status: 200,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      },
+    );
+  } catch (error) {
+    console.error("Error processing request:", error);
+    return new NextResponse(
+      JSON.stringify({ error: "Internal Server Error" }),
+      {
+        status: 500,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      },
+    );
+  }
+}
+
+// export async function GET(req: NextRequest) {
+//   try {
+//     const url = new URL(req.url);
+//     const storyId = url.searchParams.get("storyId");
+
+//     if (!storyId) {
+//       return new NextResponse(
+//         JSON.stringify({ error: "storyId is required" }),
+//         {
+//           status: 400,
+//           headers: { "Content-Type": "application/json" },
+//         },
+//       );
+//     }
+
+//     const comments = await prisma.comment.findMany({
+//       where: { storyId: storyId },
+//       include: {
+//         replies: true, // Including nested replies in the response
+//       },
+//     });
+
+//     return new NextResponse(JSON.stringify({ comments }), {
+//       status: 200,
+//       headers: { "Content-Type": "application/json" },
+//     });
+//   } catch (error) {
+//     console.error("Error processing request:", error);
+//     return new NextResponse(
+//       JSON.stringify({ error: "Internal Server Error" }),
+//       {
+//         status: 500,
+//         headers: { "Content-Type": "application/json" },
+//       },
+//     );
+//   }
+// }
+
+export async function GET(req: NextRequest) {
+  try {
+    const url = new URL(req.url);
+    const storyId = url.searchParams.get("storyId");
+    const userId = url.searchParams.get("userId");
```

Comment:
can you return an error 400 if there is no story id and no user id?

Reply:
solved